src/ToolingSharedLibrary/Includes/VbsEnclaveABI/Shared/ConversionHelpers.h

```diff
@@ -0,0 +1,543 @@
+    concept StdArray = is_std_array_v<T>;
+
+    template<typename T>
+    concept Container = is_container_v<T>;
+
+    template<typename T>
+    concept RawPtr = std::is_pointer_v<T>;
+
+    template<typename T>
+    concept NonWrappedStruct =
+        std::is_class_v<std::decay_t<T>> &&    // Must be a class or struct
+        !UniquePtr<std::decay_t<T>> &&   // Must not be unique ptr
+        !Optional<std::decay_t<T>> && // Must not be optional
+        !Container<std::decay_t<T>>; // Must not be wstring, string, vector or array
+
+    template <typename T, typename U>
+    concept IsOptionalAndUniquePtr =
+        (Optional<T> && UniquePtr<U>) ||
+        (Optional<U> && UniquePtr<T>);
+
+    template <typename T, typename U>
+    concept IsUniquePtrAndNonWrappedStruct =
+        (UniquePtr<T> && NonWrappedStruct<U>) ||
+        (UniquePtr<U> && NonWrappedStruct<T>);
+
+    template <typename T, typename U>
+    concept IsRawPtrAndUniquePtr =
+        (RawPtr<T> && UniquePtr<U>) ||
+        (RawPtr<U> && UniquePtr<T>);
+
+    template <typename T>
+    concept IsRawPtrAndNonWrappedStruct = RawPtr<T> && NonWrappedStruct<remove_pointer_t<T>>;
+
+    template <typename T>
+    concept IsRawPtrAndNotNonWrappedStruct = RawPtr<T> && !NonWrappedStruct<remove_pointer_t<T>>;
+
+    template <typename T, typename U>
+    concept IsVectorAndStdArray = (Vector<T> && StdArray<U>) || (Vector<U> && StdArray<T>);
+
+    template <typename T, typename U>
+    concept AreBothUniquePtrs = UniquePtr<T> && UniquePtr<U>;
+
+    template <typename T, typename U>
+    concept AreBothArithmeticTypes = std::is_arithmetic_v<T> && std::is_arithmetic_v<U>;
+
+    template <typename T, typename U>
+    concept AreBothEnums = std::is_enum_v<T> && std::is_enum_v<U>;
+
+    template <typename T, typename U>
+    concept AreBothTheSame = std::is_same_v<T, U>;
+
+    template <typename T, typename U>
+    concept AreBothNonWrappedStructs = NonWrappedStruct<T> && NonWrappedStruct<U>;
+
+    template <typename T, typename U>
+    concept AreBothVectors = Vector<T> && Vector<U>;
+
+    // Used only for static_asserts
+    template<typename...> struct always_false : std::false_type {};
+
+    template<ConversionType ConversionT, typename Src, typename Target>
+    decltype(auto) ConvertType(const Src& src)
+    {
+        using DecayedSrc = std::decay_t<Src>;
+        using DecayedTarget = std::decay_t<Target>;
+
+        if constexpr (ConversionT == ConversionType::ToDevType)
+        {
+            return ConvertToDevType<DecayedSrc, DecayedTarget>(src);
+        }
+        else
+        {
+            return ConvertToFlatbuffer<DecayedSrc, DecayedTarget>(src);
+        }
+    }
+
+    template <
+        template<typename...> class Wrapper,
+        typename Src,
+        typename TargetInnerType,
+        typename AllocFunc, 
+        typename ConvertFunc
+    >
+    auto ConvertToWrapper(const Src& src, AllocFunc&& allocator_func, ConvertFunc&& converter_func)
```

Comment:
>AllocFunc&& allocator_func, ConvertFunc&& converter_func [](http://example.com/codeflow?start=42&length=56)

why does this take 2 lambdas if they are always composed as,
    allocator_func(converter_func(x))
?